src/colmap/util/threading.h

```diff
@@ -38,10 +38,18 @@
 #include <list>
 #include <queue>
 #include <thread>
+#include <type_traits>
 #include <unordered_map>
 
 namespace colmap {
 
+template <class func_t, class... args_t>
+#ifdef __cpp_lib_is_invocable
+using result_of_t = std::invoke_result_t<func_t, args_t...>;
+#else
+using result_of_t = typename std::result_of<func_t(args_t...)>::type;
```

Comment:
using in the header will make this name available in any file that includes this header. I suggest to move this under the ThreadPool class below. thx